internal/pkg/api/cloudstack.go

```diff
@@ -0,0 +1,242 @@
+package api
+
+import (
+	"fmt"
+	"os"
+
+	"sigs.k8s.io/yaml"
+
+	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/templater"
+)
+
+type CloudStackConfig struct {
+	datacenterConfig    *v1alpha1.CloudStackDatacenterConfig
+	cpMachineConfig     *v1alpha1.CloudStackMachineConfig
+	workerMachineConfig *v1alpha1.CloudStackMachineConfig
+	etcdMachineConfig   *v1alpha1.CloudStackMachineConfig
+}
+
+type CloudStackFiller func(config CloudStackConfig)
+
+func AutoFillCloudStackProvider(filename string, fillers ...CloudStackFiller) ([]byte, error) {
+	var etcdMachineConfig *v1alpha1.CloudStackMachineConfig
+	// only to get name of control plane and worker node machine configs
+	clusterConfig, err := v1alpha1.GetClusterConfig(filename)
```

Comment:
you can use `v1alpha1.GetAndValidateClusterConfig` to avoid the validations below.

Reply:
actually you might just use `cluster.ParseConfigFromFile` to get the `Config` object that has everything you need in the `CloudStackConfig`

Reply:
done

Reply:
updated to use v1alpha1.GetAndValidateClusterConfig.
confg returned from cluster.ParseConfigFromFile does not have cloudstack info. 
```
type Config struct {
	Cluster               *anywherev1.Cluster
	VSphereDatacenter     *anywherev1.VSphereDatacenterConfig
	DockerDatacenter      *anywherev1.DockerDatacenterConfig
	SnowDatacenter        *anywherev1.SnowDatacenterConfig
	VSphereMachineConfigs map[string]*anywherev1.VSphereMachineConfig
	SnowMachineConfigs    map[string]*anywherev1.SnowMachineConfig
	OIDCConfigs           map[string]*anywherev1.OIDCConfig
	AWSIAMConfigs         map[string]*anywherev1.AWSIamConfig
	GitOpsConfig          *anywherev1.GitOpsConfig
	FluxConfig            *anywherev1.FluxConfig
}
```

Does Cloudstack miss a field under Config?

Reply:
yes it is new. So you will need to add cloudstack field in this Config struct